src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -294,6 +293,61 @@ public bool IsSolutionAvailable
             }
         }
 
+        public IEnumerable<string> GetDeferredProjectsFilePath()
```

Comment:
Blocking a thread is never a good idea. And should be avoided at all costs. Based on my experience in resolving deadlocks this is not recommended practice. It is justified only as a workaround when fitting asynchronous code in an event handler or a legacy interface method.